src/Components/v2/Carousel.tsx

```diff
@@ -80,40 +108,38 @@ export const LargeCarousel = (props: Props) => {
   }
 
   return (
-    <Flex
-      flexDirection="row"
-      justifyContent="space-around"
-      alignItems="center"
-      height={props.height}
-    >
-      {props.renderLeftArrow ? (
-        props.renderLeftArrow({
-          Arrow: LeftArrow,
-          getSlick: () => slickRef,
-          ...props,
-        })
-      ) : (
-        <LeftArrow />
-      )}
-
-      <CarouselContainer height={props.height}>
-        <Slick {...slickConfig} ref={slider => (slickRef = slider)}>
-          {props.data.map((slide, index) => {
-            return <Box key={index}>{props.render(slide)}</Box>
-          })}
-        </Slick>
-      </CarouselContainer>
-
-      {props.renderRightArrow ? (
-        props.renderRightArrow({
-          Arrow: RightArrow,
-          getSlick: () => slickRef,
-          ...props,
-        })
-      ) : (
-        <RightArrow />
-      )}
-    </Flex>
+    <Box position="relative">
+      <Flex
+        flexDirection="row"
+        justifyContent="space-around"
+        alignItems="center"
+        height={props.height}
+      >
+        {props.renderLeftArrow
+          ? props.renderLeftArrow({
+              Arrow: LeftArrow,
+              getSlick: () => slickRef,
+              ...props,
+            })
+          : !props.firstSlideVisible && <LeftArrow />}
+
+        <CarouselContainer height={props.height}>
+          <Slick {...slickConfig} ref={slider => (slickRef = slider)}>
+            {props.data.map((slide, index) => {
+              return <Box key={index}>{props.render(slide)}</Box>
+            })}
+          </Slick>
+        </CarouselContainer>
+
+        {props.renderRightArrow
+          ? props.renderRightArrow({
+              Arrow: RightArrow,
+              getSlick: () => slickRef,
+              ...props,
+            })
+          : !props.lastSlideVisible && <RightArrow />}
```

Comment:
Rather than extending the api, can pass state into callback up above and can compute these things based on the current slide _index_. In slide renderArrow callbacks: 

```
if (slideIndex === 0) {
  // hide arrow
}

if (slideIndex = totalSlides - 1) {
  // hide right arrow 
}